crates/sui-core/src/authority.rs

```diff
@@ -704,6 +704,11 @@ impl AuthorityState {
             .instrument(span)
             .await?;
 
+        if self.is_cert_awaiting_sequencing(digest)? {
```

Comment:
I wonder if adding the check here is sufficient? i.e. whether for every path that leads to here we have already verified the cert

Reply:
This is intentionally pre-verification. It's okay to ask if a cert has been sequenced without first verifying the signatures.